File: app/src/main/java/org/nitri/opentopo/nearby/entity/NearbyItem.java
package org.nitri.opentopo.nearby.entity;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;

@Entity(tableName = "Nearby", indices={@Index(value = "pageid", unique = true)})
public class NearbyItem {

    @PrimaryKey(autoGenerate = true)
    private int id;

    private String pageid;

    private int index;

    private String title;

    private String description;

    private String thumbnail;

    private int width;

    private int height;

    private double lat;

    private double lon;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPageid() {
        return pageid;
    }

    public void setPageid(String pageid) {
        this.pageid = pageid;
    }

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public double getLat() {
        return lat;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public double getLon() {
        return lon;
    }

    public void setLon(double lon) {
        this.lon = lon;
    }
}

File: app/src/main/java/org/nitri/opentopo/nearby/entity/NearbyItem.kt
package org.nitri.opentopo.nearby.entity

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "Nearby", indices = [Index(value = arrayOf("pageid"), unique = true)])
class NearbyItem : Comparable<NearbyItem> {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    var id = 0
    @JvmField
    var pageid: String? = null
    @JvmField
    var index = 0
    @JvmField
    var title: String? = null
    @JvmField
    var description: String? = null
    @JvmField
    var thumbnail: String? = null
    @JvmField
    var width = 0
    @JvmField
    var height = 0
    @JvmField
    var lat = 0.0
    @JvmField
    var lon = 0.0
    @JvmField
    var url: String? = null

    @JvmField
    @Ignore
    var distance = 0
    override fun compareTo(other: NearbyItem): Int {
        return Integer.compare(distance, other.distance)
    }
}
